packages/catalog-process/src/services/catalogService.ts

```diff
@@ -578,10 +592,11 @@ export function catalogServiceBuilder(
       logger.info(`Deleting EService ${eserviceId}`);
 
       const eservice = await retrieveEService(eserviceId, readModelService);
-      assertRequesterAllowed(eservice.data.producerId, authData);
-
+      assertRequesterIsProducer(eservice.data.producerId, authData);
       assertIsDraftEservice(eservice.data);
 
+      await assertNoValidDelegationAssociated(eserviceId, readModelService);
```

Comment:
This is returning `operationForbidden` if this eservice has a delegation. But in that case, how the caller could understand why? Isn't it better to have a specific error for this?

Reply:
Yes, makes sense, let's add a custom error